File: VB/SchedulerAPISample/CodeExamples/Services.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing
Imports DevExpress.XtraScheduler.Services
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace SchedulerAPISample.CodeExamples
    Friend Class Services
        Private Shared Sub AppointmentFormatStringServiceAction(ByVal scheduler As SchedulerControl)
'            #Region "#AppointmentFormatStringService"
            Dim oldService As IAppointmentFormatStringService = scheduler.GetService(Of IAppointmentFormatStringService)()
            If oldService IsNot Nothing Then
                Dim newService As New MyAppointmentFormatStringService(oldService)
                scheduler.RemoveService(GetType(IAppointmentFormatStringService))
                scheduler.AddService(GetType(IAppointmentFormatStringService), newService)
            End If
            scheduler.ActiveView.LayoutChanged()
'            #End Region ' #AppointmentFormatStringService
        End Sub

        #Region "#@AppointmentFormatStringService"
        Public Class MyAppointmentFormatStringService
            Inherits AppointmentFormatStringServiceWrapper

            Public Sub New(ByVal service As IAppointmentFormatStringService)
                MyBase.New(service)
            End Sub
            Public Overrides Function GetVerticalAppointmentStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm:ss - 'VerticalAppointmentStart'}"
            End Function
            Public Overrides Function GetVerticalAppointmentEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm:ss - 'VerticalAppointmentEnd'}"
            End Function
            Public Overrides Function GetHorizontalAppointmentEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm - 'HorizontalAppointmentEnd'}"
            End Function
            Public Overrides Function GetHorizontalAppointmentStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm - 'HorizontalAppointmentStart'}"
            End Function
            Public Overrides Function GetContinueItemStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm MMM dd - 'ContinueItemStart'}"
            End Function
            Public Overrides Function GetContinueItemEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
                Return "{0: HHmm MMM dd - 'ContinueItemEnd'}"
            End Function
        End Class
        #End Region ' #@AppointmentFormatStringService
    End Class
End Namespace

File: CS/SchedulerAPISample/CodeExamples/Services.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;
using DevExpress.XtraScheduler.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SchedulerAPISample.CodeExamples
{
    class Services
    {
        static void AppointmentFormatStringServiceAction(SchedulerControl scheduler)
        {
            #region #AppointmentFormatStringService
            IAppointmentFormatStringService oldService = scheduler.GetService<IAppointmentFormatStringService>();
            if (oldService != null)
            {
                MyAppointmentFormatStringService newService = new MyAppointmentFormatStringService(oldService);
                scheduler.RemoveService(typeof(IAppointmentFormatStringService));
                scheduler.AddService(typeof(IAppointmentFormatStringService), newService);
            }
            scheduler.ActiveView.LayoutChanged();
            #endregion #AppointmentFormatStringService
        }

        #region #@AppointmentFormatStringService
        public class MyAppointmentFormatStringService : AppointmentFormatStringServiceWrapper
        {
            public MyAppointmentFormatStringService(IAppointmentFormatStringService service)
                : base(service) { }
            public override string GetVerticalAppointmentStartFormat(
                IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm:ss - 'VerticalAppointmentStart'}";
            }
            public override string GetVerticalAppointmentEndFormat(
                IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm:ss - 'VerticalAppointmentEnd'}";
            }
            public override string GetHorizontalAppointmentEndFormat(
                IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm - 'HorizontalAppointmentEnd'}";
            }
            public override string GetHorizontalAppointmentStartFormat(
                IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm - 'HorizontalAppointmentStart'}";
            }
            public override string GetContinueItemStartFormat(IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm MMM dd - 'ContinueItemStart'}";
            }
            public override string GetContinueItemEndFormat(IAppointmentViewInfo aptViewInfo)
            {
                return "{0: HHmm MMM dd - 'ContinueItemEnd'}";
            }
        }
        #endregion #@AppointmentFormatStringService
    }
}
